src/Domain/CoordinateMaps/TimeDependent/Shape.cpp

```diff
@@ -398,6 +376,15 @@ void Shape::pup(PUP::er& p) {
   if (p.isUnpacking()) {
     ylm_ = ylm::Spherepack(l_max_, m_max_);
   }
+
+  if (version >= 1) {
+    p | f_of_t_names_;
+  } else if (p.isUnpacking()) {
+    f_of_t_names_ = std::unordered_set<std::string>{{shape_f_of_t_name_}};
```

Comment:
Does this pattern of serializing the names work correctly? I think some names are still hardcoded, so if they're changed in the code and the old name is deserialized then there's an inconsistency between the hardcoded name and the name in the set. This means either you don't serialize the names at all but just return the hardcoded names (no need for the member variable even) and don't bother with serialization. Then the serialization will still be broken if names are changed because the domain/maps are inconsistent with the serialized FoT map in the file, but that's also the case on develop right now I think. Or you fix all this by serializing the hardcoded FoT name in the maps. Does that make sense or am I misunderstanding something?

Reply:
Conclusions of discussion with @nilsvu offline:

The "hardcoded" names are actually in the control system. The control system expects certain names in the functions of time (and the domain creators then must supply those names to be usable with control system). Meaning if we change the control system names (hardcoded in `src/ControlSystems`), then everything breaks. (so let's not do that)

The maps themselves actually take names in their constructors and already serialize them. So what's important here is that when we deserialize the maps and the functions of time, everything is consistent. So since we already serialize the names in the maps, we can just always reconstruct the `std::unordered_set` that is needed now from the serialized names and we don't need to change the versions of the pup functions. So I'll make a change to not serialize the `std::unordered_set` and only reconstruct it.